From ac74b9067d079b03f3fe4236270f9eb34121009b Mon Sep 17 00:00:00 2001
From: Fam Zheng <famz@redhat.com>
Date: Thu, 12 Oct 2017 13:54:45 +0200
Subject: [PATCH 16/69] osdep: Force define F_OFD_GETLK (RHEL only)

RH-Author: Fam Zheng <famz@redhat.com>
Message-id: <20171012135445.4214-1-famz@redhat.com>
Patchwork-id: 77220
O-Subject: [RHV7.5 qemu-kvm-ma PATCH] osdep: Force define F_OFD_GETLK (RHEL only)
Bugzilla: 1378241
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>

BZ: 1378241
Brew: https://brewweb.engineering.redhat.com/brew/taskinfo?taskID=14254218

glibc is not ready yet (BZ 1461231, which is deferred to 7.6 due to
capacity), so the OFD constants are not defined in the system headers we
pull in. (They do exist in the headers of latest kernel-headers package,
but we don't want to include that anyway.)

Actually the constants are all that are missing before we can call image
locking done in 7.5, so there is no reason to wait for glibc.

This patch can be reverted once the new glibc headers are in place.

Signed-off-by: Fam Zheng <famz@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 util/osdep.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/util/osdep.c b/util/osdep.c
index a479fed..8358a44 100644
--- a/util/osdep.c
+++ b/util/osdep.c
@@ -23,6 +23,11 @@
  */
 #include "qemu/osdep.h"
 
+#ifndef F_OFD_SETLK
+#define F_OFD_GETLK    36
+#define F_OFD_SETLK    37
+#endif
+
 /* Needed early for CONFIG_BSD etc. */
 
 #ifdef CONFIG_SOLARIS
-- 
1.8.3.1

